### ISSUE TITLE
Enable backup option for azurerm_recovery_services_protected_vm

### ISSUE THREAD

**Issue by mbrancato** [terraform-providers/terraform-provider-azurerm#2279]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.11.10
+ provider.azurerm v1.17.0
```

### Affected Resource(s)

* azurerm_recovery_services_protected_vm

### Terraform Configuration Files

```hcl
resource "azurerm_recovery_services_protected_vm" "protected" {
  resource_group_name = "${azurerm_resource_group.rg.name}"
  recovery_vault_name = "${azurerm_recovery_services_vault.vault.name}"
  source_vm_id        = "${azurerm_virtual_machine.vm.id}"
  backup_policy_id    = "${azurerm_recovery_services_protection_policy_vm.default.id}"
}
```

### Expected Behavior

Protection would be enabled after creating the protected VM resource.

### Actual Behavior

Protection must be enabled via the Portal or CLI, etc. The `azurerm_recovery_services_protected_vm` does not seem to actually have an option to enable the backup. It may be because the `ProtectionStatus` is not set.

### References

https://github.com/terraform-providers/terraform-provider-azurerm/blob/fdead9edae5316010d29cdc5ef080f6efdb43131/azurerm/resource_arm_recovery_services_protected_vm.go#L92-L99

**Comment by katbyte** [comment]
Hi @mbrancato,

I have looked at the `ProtectionStatus` property and [it appears](https://github.com/Azure/azure-docs-powershell/blob/master/azurermps-4.4.1/AzureRM.RecoveryServices.Backup/Get-AzureRmRecoveryServicesBackupItem.md) its only valid values are `Healthy` or `Unhealthy`, and that doesn't seem like it should be set, or enable the backup, yet the documentation indicates it can be set?

Would you mind sharing the CLI command that you use to enable the backup?

**Comment by mbrancato** [comment]
hi @katbyte 

The Azure CLI command I think would be `az backup protection enable-for-vm`.

However, I think what I'm dealing with is some incompatiability with VMs deployed without the `provision_vm_agent` set, even after the VM agent was manually installed post-creation. I'm working with my team to reach out to Microsoft on this, but will update back here as all our VMs were deployed with Terraform. The problem seems to be that no VM agent extensions seem to be installed.

**Comment by mbrancato** [comment]
Thanks @katbyte, I talked to support and solved the issue. Closing and referencing the necessary fix in #2298
